### PR TITLE
Core/require as babel plugin

### PR DESCRIPTION
**Pull request by mrkno**
Make require hooks a babel plugin.

### CODE REVIEW

File: core/require.js [hunk 1]
```diff
@@ -54,24 +53,22 @@ global.requireHook = function (req) {
...
+module.exports = function () {
+    return {
+        visitor: {
+            Program(path) {
+                path.unshiftContainer('body', babylon.parse('require = (global || GLOBAL).requireHook(require);').program.body[0]);
+            }
+        }
+    };
+};
+exports.default = module.exports;
```
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Newline required at end of file but not found.](https://www.codacy.com/app/theprojectshop/Kassy/file/2651193026/issues/source?bid=3282987&fileBranchId=3324570#l74)

Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Newline required at end of file but not found.](https://www.codacy.com/app/theprojectshop/Kassy/file/2651207256/issues/source?bid=3282987&fileBranchId=3324570#l74)

File: core/prototypes.js [hunk 2]
```diff
@@ -50,103 +37,3 @@ if (typeof String.prototype.capitiliseFirst != 'function') {
 		return this;
 	};
 }
-
-// array.includes
-if (!Array.prototype.includes) {
```
Comment (Awarua-) [inline core/prototypes.js]: Are we sure that nothing depends on these functions?

File: modules/8ball/8Ball.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
-var request = require.safe('request');
-
-insults = ['You\'re asking the impossible',
+var request = require.safe('request'),
```
Comment (Awarua-) [inline modules/8ball/8Ball.js]: Not sure why this is here?

**Comment by Awarua-** [comment]
Missing version bump

**Comment by Awarua-** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/concierge/Kassy/pull-request/164/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=concierge/Kassy)

File: core/core_modules/help.js [hunk 4]
```diff
@@ -63,15 +64,14 @@ exports.run = function(api, event) {
...
-    if (commands.length === 1) {
-        help = shortSummary.call(exports.platform, context);
-    }
-    else {
-        commands.splice(0, 1);
-        help = longDescription.call(exports.platform, commands.join(' '), context);
-    }
+		help;
+	if (commands.length === 1) {
+		help = shortSummary.call(exports.platform, context, event);
```
Comment (mrkno) [inline core/core_modules/help.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361708/issues/source?bid=3282987&fileBranchId=3324570#l69)

File: core/core_modules/help.js [hunk 5]
```diff
@@ -63,15 +64,14 @@ exports.run = function(api, event) {
...
+	}
+	else {
+		commands.splice(0, 1);
+		help = longDescription.call(exports.platform, commands.join(' '), context, event);
```
Comment (mrkno) [inline core/core_modules/help.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361708/issues/source?bid=3282987&fileBranchId=3324570#l73)

File: core/core_integrations/slack.js [hunk 6]
```diff
@@ -274,6 +270,35 @@ var request = require.safe('request'),
...
+        console.debug('slack-> Connecting to socket');
+
+        if (connectionDetails) {
+            socket = new WebSocket(connectionDetails.url);
+            console.debug(connectionDetails.url);
+            sockets[connectionDetails.team_id] = socket;
+            socket.on('open', function() {
+                console.debug('slack-> Connection to team: ' + teamData[connectionDetails.team_id].team.name + ' established');
+            }).on('message', function(data) {
+                startPingPongTimer(connectionDetails.team_id);
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: ["startPingPongTimer" was used before it was defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361702/issues/source?bid=3282987&fileBranchId=3324570#l284)

File: core/core_integrations/slack.js [hunk 7]
```diff
@@ -274,6 +270,35 @@ var request = require.safe('request'),
...
+                eventReceived(JSON.parse(data), connectionDetails.team_id);
+            }).on('ping', function() {
+                console.debug('slack-> Recieved ping, sending pong');
+                sendSocketMessage('pong', connectionDetails.team_id);
+            }).on('pong', function() {
+                pong(connectionDetails.team_id);
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: ["pong" was used before it was defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361702/issues/source?bid=3282987&fileBranchId=3324570#l290)

File: core/core_integrations/slack.js [hunk 8]
```diff
@@ -274,6 +270,35 @@ var request = require.safe('request'),
...
+
+        if (connectionDetails) {
+            socket = new WebSocket(connectionDetails.url);
+            console.debug(connectionDetails.url);
+            sockets[connectionDetails.team_id] = socket;
+            socket.on('open', function() {
+                console.debug('slack-> Connection to team: ' + teamData[connectionDetails.team_id].team.name + ' established');
+            }).on('message', function(data) {
+                startPingPongTimer(connectionDetails.team_id);
+                eventReceived(JSON.parse(data), connectionDetails.team_id);
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: ["eventReceived" was used before it was defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361702/issues/source?bid=3282987&fileBranchId=3324570#l285)

File: modules/common/wikiquote.js [hunk 9]
```diff
@@ -7,135 +7,139 @@ var request = require.safe('request');
...
 
     var chooseQuote = function(quotes) {
-      var randomNum = Math.floor(Math.random()*quotes.quotes.length);
+      var randomNum = Math.floor(Math.random() * quotes.quotes.length);
       success({ titles: quotes.titles, quote: quotes.quotes[randomNum] });
     };
 
     var getQuotes = function(pageId, sections) {
-      var randomNum = Math.floor(Math.random()*sections.sections.length);
+      var randomNum = Math.floor(Math.random() * sections.sections.length);
```
Comment (mrkno) [inline modules/common/wikiquote.js]: Codacy Issue found: [Expected indentation of 8 space characters but found 6.](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361737/issues/source?bid=3282987&fileBranchId=3324570#l142)

File: modules/common/wikiquote.js [hunk 10]
```diff
@@ -7,135 +7,139 @@ var request = require.safe('request');
...
-        }
-      });
-  };
+                for (var p in pages) {
+                    var page = pages[p];
+                    // api can return invalid recrods, these are marked as "missing"
+                    if(!('missing' in page)) {
+                        pageId = page.pageid;
+                        break;
+                  }
```
Comment (mrkno) [inline modules/common/wikiquote.js]: Codacy Issue found: [Expected indentation of 20 space characters but found 18.](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361737/issues/source?bid=3282987&fileBranchId=3324570#l36)

File: modules/common/wikiquote.js [hunk 11]
```diff
@@ -7,135 +7,139 @@ var request = require.safe('request');
...
+    */
+    exports.getRandomQuote = function(titles, success, error) {
 
     var errorFunction = function(msg) {
       error(msg);
     };
 
     var chooseQuote = function(quotes) {
-      var randomNum = Math.floor(Math.random()*quotes.quotes.length);
+      var randomNum = Math.floor(Math.random() * quotes.quotes.length);
```
Comment (mrkno) [inline modules/common/wikiquote.js]: Codacy Issue found: [Expected indentation of 8 space characters but found 6.](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361737/issues/source?bid=3282987&fileBranchId=3324570#l137)

File: core/core_integrations/facebook.js [hunk 12]
```diff
@@ -127,35 +127,35 @@ exports.start = function(callback) {
...
+                    var usrs = event.logMessageData.removed_participants;
+                    for (var i = 0; i < usrs.length; i++) {
+                        usrs[i] = usrs[i].split(':')[1];
+                        if (threadInfo[event.threadID] && threadInfo[event.threadID][usrs[i]]) {
+                            delete threadInfo[event.threadID][usrs[i]];
                         }
                     }
                     break;
+                    }
+                    case 'log:subscribe': {
```
Comment (mrkno) [inline core/core_integrations/facebook.js]: Codacy Issue found: [Expected indentation of 20 space characters but found 24.](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361698/issues/source?bid=3282987&fileBranchId=3324570#l149)

File: core/core_integrations/facebook.js [hunk 13]
```diff
@@ -127,35 +127,35 @@ exports.start = function(callback) {
...
+                case 'log:unsubscribe': {
+                    var usrs = event.logMessageData.removed_participants;
+                    for (var i = 0; i < usrs.length; i++) {
+                        usrs[i] = usrs[i].split(':')[1];
+                        if (threadInfo[event.threadID] && threadInfo[event.threadID][usrs[i]]) {
+                            delete threadInfo[event.threadID][usrs[i]];
                         }
                     }
                     break;
+                    }
```
Comment (mrkno) [inline core/core_integrations/facebook.js]: Codacy Issue found: [Expected indentation of 16 space characters but found 20.](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361698/issues/source?bid=3282987&fileBranchId=3324570#l148)

**Comment by Awarua-** [comment]
Approved

**Pull request merged**